fabtests/pytest/efa/test_unexpected_msg.py

```diff
@@ -1,7 +1,13 @@
 import pytest
+from efa.efa_common import efa_run_client_server_test
 
+@pytest.mark.parametrize("iteration_type",
+                         [pytest.param("short", marks=pytest.mark.short),
+                          pytest.param("standard", marks=pytest.mark.standard)])
 @pytest.mark.functional
-def test_unexpected_msg(cmdline_args):
-    from common import ClientServerTest
-    test = ClientServerTest(cmdline_args, "fi_unexpected_msg -e rdm -I 10")
-    test.run()
+def test_unexpected_msg(cmdline_args, completion_semantic, memory_type, message_size):
+    if completion_semantic == "delivery_complete":
+        pytest.skip("Unexpected message test does not support delivery_complete completion semantic")
+
+    efa_run_client_server_test(cmdline_args, "fi_unexpected_msg", iteration_type,
```

Comment:
why turn off data check?

Reply:
Test failed with unrecognized option when I passed in `-v`, so I thought the test doesn't support data verification. But it actually does. I'm having other issues with the HMEM buffer though. So making this PR a draft for now.

Reply:
Data check is back